### PR TITLE
[epskc] add support for multiple ephemeral key callbacks

### PR DESCRIPTION
**Pull request by sunytt**

### CODE REVIEW

File: src/border_agent/border_agent.cpp [hunk 1]
```diff
@@ -320,6 +325,11 @@ void BorderAgent::UnpublishEpskcService()
     });
 }
 
+void BorderAgent::AddEphemeralKeyChangedCallback(EphemeralKeyChangedCallback aCallback)
+{
+    mEphemeralKeyChangedCallbacks.emplace_back(std::move(aCallback));
```
Comment (superwhd) [inline src/border_agent/border_agent.cpp:330]:
```suggestion
    mEphemeralKeyChangedCallbacks.push_back(std::move(aCallback));
```
nit: no need to use `emplace_back`.

File: src/border_agent/border_agent.hpp [hunk 2]
```diff
@@ -144,6 +147,14 @@ class BorderAgent : private NonCopyable
      */
     void HandleMdnsState(Mdns::Publisher::State aState);
 
+    /**
+     * This method adds a event listener for ephemeral key changes.
```
Comment (superwhd) [inline src/border_agent/border_agent.hpp:151]:
```suggestion
     * This method adds a callback for ephemeral key changes.
```

**Review by superwhd — APPROVED** [review]
LGTM overall

File: src/border_agent/border_agent.hpp (hunk 2 above)
Comment (sunytt) [reply src/border_agent/border_agent.hpp]: Done

File: src/border_agent/border_agent.cpp (hunk 1 above)
Comment (sunytt) [reply src/border_agent/border_agent.cpp]: Done

**Review by Irving-cl — APPROVED** [review]
LGTM

**Comment by sunytt** [comment]
Removing from reviewer list as @librasungirl is OOO for a week.

**Review by abtink — APPROVED** [review]
👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**